File: mabi/media/effect.js
/**
 * @param options {
 *   condition: 発動条件
 *     条件がない場合は省略
 *   op:
 *   param:
 *   min:
 *   max: ゆらぎ幅が無い場合は省略する
 * }
 */
mabi.Effect = function(options){
    options = options || {
    };

    this.condition = options.condition;
    this.op_ = options.op;
    this.param_ = options.param;
    this.min_ = options.min;
    this.max_ = options.max;
};

mabi.Effect.prototype.condition = function(){
    return this.condition_;
};

mabi.Effect.prototype.op = function(){
    return this.op_;
};

mabi.Effect.prototype.param = function(){
    return this.param_;
};

mabi.Effect.prototype.min = function(){
    return this.min_;
};

mabi.Effect.prototype.max = function(){
    return this.max_;
};

mabi.Effect.prototype.paramText = function(){
    return this.param_;
};

/**
 * TODO 効果がプラス効果かどうか調べる
 */
mabi.Effect.prototype.plus = function(){
    return true;
};

File: mabi/effect.py
# -*- coding: utf-8 -*-

class Effect:
    '''各種効果を表す
    '''

    def __init__(self, text=None, op='+', min=None, max=None, param=None, condition=''):
        if text:
            a = text.split(',')
            self.op = a[1]
            self.param = a[0]
            self.min = int(a[2])
            self.max= int(a[3])
            if len(a) >= 5: 
                self.condition = a[4]
            else:
                self.condition = ''
        else:
            if max == None: max = min

            self.op = op
            self.param = param
            self.min = int(min)
            self.max = int(max)
            self.condition = condition

    def __unicode__(self):
        ''''''
        return u'%s,%s,%s,%s,%s' % (
            self.param, self.op, self.min, self.max, self.condition)

    def __repr__(self):
        ''''''
        return u'%s,%s,%s,%s,%s' % (
            self.param, self.op, self.min, self.max, self.condition)


    def __eq__(self, other):
        a = (isinstance(other, Effect) and
             self.op == other.op and
             self.min == other.min and
             self.max == other.max and
             self.param == other.param and
             self.condition == other.condition)
        return a
            
    def __ne__(self, other):
        return not self.__eq__(other)

